src/main/java/frc/robot/subsystems/climb/Climb.java

```diff
@@ -0,0 +1,220 @@
+    private double _desiredHeight;
+
+    // Finds the value that the robot needs to adjust by in order to be level, based on the gyro angle
+    private double _heightAdjustment = 0.0;
+
+    private Double _leftHeight;
+    private Double _rightHeight;
+
+    private Double _climbSetpointLeft  = null;
+    private Double _climbSetpointRight = null;
+
+    private double _climbMinExtension  = Constants.Climb.MIN_EXTENSION; // TODO: tune value
+    private double _climbMaxExtension  = Constants.Climb.MAX_EXTENSION; // TODO: tune value
+
+    public Climb(ClimbIO io)
+    {
+        _io = io;
+
+        _climbFeedbackLeft  = new PIDController(0, 0, 0); // TODO: tune values
+        _climbFeedbackRight = new PIDController(0, 0, 0);
+
+        _tiltPID  = new PIDController(0, 0, 0); // TODO: tune values
+        _leftPID  = new PIDController(0, 0, 0);
+        _rightPID = new PIDController(0, 0, 0);
+
+        _gyro = new AHRS(Port.kMXP);
+
+        // Takes the desired position of both arms, and takes the needed adjustment calculated from the tilt to set the position - setpoint(?) for both arms
+        _leftHeight  = _desiredHeight + _heightAdjustment;
+        _rightHeight = _desiredHeight - _heightAdjustment;
+    }
+
+    @Override
+    public void periodic()
+    {
+        _io.updateInputs(_inputs);
+        Logger.processInputs("Climb", _inputs);
+
+        if (_climbSetpointLeft != null)
+        {
+            _io.setVoltageLeft(_climbFeedbackLeft.calculate(getExtensionLeft(), _climbSetpointLeft));
+        }
+
+        if (_climbSetpointRight != null)
+        {
+            _io.setVoltageRight(_climbFeedbackRight.calculate(getExtensionRight(), _climbSetpointRight));
+        }
+
+        _heightAdjustment = _tiltPID.calculate(getCurrentTilt(), 0);
+
+        // Gets the current position of the left and the right arms, and sets it to the desired position based on the tilt input, also applies this voltage
+        if(_leftHeight != null)
+        {
+            _io.setAlgorithmVoltageLeft(_leftPID.calculate(getExtensionLeft(), _leftHeight));
+        }
+
+        if(_rightHeight != null)
+        {
+            _io.setAlgorithmVoltageRight(_rightPID.calculate(getExtensionRight(), _rightHeight));
+        }
+    }
+
+    public void stop()
+    {
+        _io.setVoltageLeft(0.0);
+        _io.setVoltageRight(0.0);
+
+        _climbSetpointLeft  = null;
+        _climbSetpointRight = null;
+
+        setLockState(true);
+    }
+
+    public void setLockState(boolean enabled)
+    {
+        _io.setLockStateLeft(enabled, _inputs);
+        _io.setLockStateRight(enabled, _inputs);
+    }
+
+    public void setSetpointLeft(double setpoint)
+    {
+        _climbSetpointLeft = MathUtil.clamp(setpoint, _climbMinExtension, _climbMaxExtension);
+    }
+    
+    public void setSetpointRight(double setpoint)
+    {
+        _climbSetpointRight = MathUtil.clamp(setpoint, _climbMinExtension, _climbMaxExtension);
+    }
```

Comment:
At the moment, I don't think there's really a desire to have the setpoints for the arms being individually settable. I think the arms effectively need to be controlled together if we're setting the height automatically, and the tilt algorithm handles this.

For that reason, I think we can remove these methods. The `setHeight` method you have below handles the need for setting the setpoint